juspay/hyperswitch
# feat(payments_update): update payment_method_billing in payment update

## Type of Change



## Description

This PR allows the `payment_method_billing` to be updated in the payments update flow.

## Motivation and Context


## How did you test it?

- Create a payment with confirm as false
```bash
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'api-key: dev_VTitmyLzKQhsrNhv6DmgKbzAxaXOLKuWkcY37HT3f7vgWvXBIoYuIg4FU61EGVF6' \
--data-raw '{
    "amount": 6540,
    "currency": "USD",
    "confirm": false,
    "customer": {
        "id": "cus_PAxm0MeCGM5TowgDKPc9",
        "email": "example@juspay.in"
    }
}'
```

- Update the payment with `payment_method_data.billing`
```bash
curl --location 'http://localhost:8080/payments/pay_62qJ2OaJTzci4aKkXjm9' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_VTitmyLzKQhsrNhv6DmgKbzAxaXOLKuWkcY37HT3f7vgWvXBIoYuIg4FU61EGVF6' \
--data '{
    "payment_method_data": {
        "billing": {
            "address": {
                "line1": "1467",
                "line2": "Harrison Street",
                "city": "San Fransico",
                "state": "California",
                "zip": "94122",
                "country": "IN",
                "first_name": "Narayan",
                "last_name": "Bhat"
            }
        }
    }
}'
```

- Retrieve the payment and ensure that the billing details are updated

## Checklist
